internal/planner/type_join.go

```diff
@@ -825,6 +827,28 @@ func (join *invertibleTypeJoin) invertJoinDirectionWithIndex(
 	return nil
 }
 
+// isOrderedByIndex checks if the plan is ordered by an index.
+func isOrderedByIndex(plan planNode) bool {
```

Comment:
suggestion: This file seems an odd location for this function, I suggest moving it somewhere else, probably to `planner.go` - the only place it is currently consumed.